tools/kwiver_tool_runner.cxx

```diff
@@ -182,11 +203,42 @@ int main(int argc, char *argv[])
     // Create applet based on the name provided
     applet_factory app_fact;
     kwiver::tools::kwiver_applet_sptr applet( app_fact.create( options.m_applet_name ) );
+
     tool_context->m_applet_name = options.m_applet_name;
+
+    // Pass the context to the applet. This is done as a separate call
+    // because the default factory for applets does not take any
+    // parameters.
     applet->initialize( tool_context.get() );
 
+    // Call the applet so it can add the commands that it is looking
+    // for.
+    applet->add_command_options();
+
+    // Convert args list back to argv style. :-(
+    // This is needed for the command options support package.
+    std::vector<char *> argv_vect( options.m_applet_args.size() + 1 );    // one extra for the null
+    for (std::size_t i = 0; i != options.m_applet_args.size(); ++i)
+    {
+      argv_vect[i] = &options.m_applet_args[i][0];
+    }
+
+    // The parse result has to be created locally due to class design.
+    // No default CTOR, copy CTOR or operation.
+    int local_argc = options.m_applet_args.size();
+    char** local_argv = &argv_vect[0];
+    cxxopts::ParseResult local_result = applet->m_cmd_options->parse( local_argc, local_argv );
+
+    // Make results available in the context,
+    tool_context->m_result = &local_result; // in this case the address of a stack variable is o.k.
```

Comment:
why not just make `m_result` a non-pointer type and assign the results directly too it?  Then you don't have all the ambiguity about who owns the memory.  Is there a reason `m_result` need to be a pointer?

Reply:
That would be preferable but there is no default CTOR, copy CTOR nor assignment operation for that class. As a result I can not copy that object. There must be some reason for this restriction so I don't want to change that behaviour.